staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/fifo_list.go

```diff
@@ -72,10 +80,29 @@ func (l *requestFIFO) Length() int {
 	return l.Len()
 }
 
+func (l *requestFIFO) Width() int {
+	l.lock.Lock()
+	defer l.lock.Unlock()
+	return l.width
+}
+
 func (l *requestFIFO) Enqueue(req *request) removeFromFIFOFunc {
+	l.lock.Lock()
+	defer l.lock.Unlock()
+
 	e := l.PushBack(req)
+	l.width += req.Seats()
+	deleted := false
+
 	return func() *request {
+		l.lock.Lock()
+		defer l.lock.Unlock()
+
 		l.Remove(e)
+		if !deleted {
```

Comment:
nit: an alternative to `deleted` is to use the `e.Value` field as a sentinel. Inside `Dequeue` we could set it to `nil` 
```
defer func() {
 l.Remove(e)
e.Value = nil
}()

```

and inside `Enqueue` we could have
```
return func() *request {
  if e.Value != nil { 
    l.Remove(e)
    e.Value = nil
  }
)
```

Reply:
done